aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
-.. include:: /../common/stub-topic.txt
+Migrating HTTP Modules to Middleware
+====================================
 
-|stub-icon| Migrating HTTP Modules to Middleware
-======================================================
+By `Matt Perdeck`_
 
-.. include:: /../common/stub-notice.txt
+This article shows how to migrate existing ASP.NET `HTTP modules and handlers <https://msdn.microsoft.com/en-us/library/bb398986.aspx>`_ to ASP.NET 5 :ref:`middleware <fundamentals-middleware>`.
 
-.. _issue: https://github.com/aspnet/Docs/issues/98
+.. contents:: In this article:
+  :local:
+  :depth: 1
+
+Handlers and modules revisited
+------------------------------
+Before proceeding to ASP.NET 5 middleware, let's first recap how HTTP modules and handlers work: 
+
+.. image:: http-modules/_static/moduleshandlers.png
+
+Handlers are:
+	* Classes that implement `IHttpHandler <https://msdn.microsoft.com/en-us/library/system.web.ihttphandler(v=vs.100).aspx>`_
+	* Used to handle requests with a given file name or extension, such as *.report*
+	* `Configured <https://msdn.microsoft.com/en-us/library/46c5ddfy(v=vs.100).aspx>`__ in the *web.config*
+
+Modules are:
+	* Classes that implement `IHttpModule <https://msdn.microsoft.com/en-us/library/system.web.ihttpmodule(v=vs.100).aspx>`_
+	* Invoked for every request
+	* Able to stop further processing of a request
+	* Able to add to the HTTP response, or create their own
+	* `Configured <https://msdn.microsoft.com/en-us/library/ms227673(v=vs.100).aspx>`__ in the web.config
```

Comment:
Chose one of these
Configured in Web.config
or
Configured in the  Web.config file
I prefer the shorter version.